In [2]:
import pandas as pd

from google.colab import files
uploaded = files.upload()

df = pd.read_csv('House Price India.csv')

df.head()

Saving House Price India.csv to House Price India.csv


,id,Date,number of bedrooms,number of bathrooms,living area,lot area,number of floors,waterfront present,number of views,condition of the house,...,Built Year,Renovation Year,Postal Code,Lattitude,Longitude,living_area_renov,lot_area_renov,Number of schools nearby,Distance from the airport,Price
0,6762810145,42491,5,2.50,3650,9050,2.0,0,4,5,...,1921,0,122003,52.8645,-114.557,2880,5400,2,58,2380000
1,6762810635,42491,4,2.50,2920,4000,1.5,0,0,5,...,1909,0,122004,52.8878,-114.470,2470,4000,2,51,1400000
2,6762810998,42491,5,2.75,2910,9480,1.5,0,0,3,...,1939,0,122004,52.8852,-114.468,2940,6600,1,53,1200000
3,6762812605,42491,4,2.50,3310,42998,2.0,0,0,3,...,2001,0,122005,52.9532,-114.321,3350,42847,3,76,838000
4,6762812919,42491,3,2.00,2710,4500,1.5,0,0,4,...,1929,0,122006,52.9047,-114.485,2060,4500,1,51,805000


## ***Basic Checks***

In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 14620 entries, 0 to 14619
Data columns (total 23 columns):
 #   Column                                 Non-Null Count  Dtype  
---  ------                                 --------------  -----  
 0   id                                     14620 non-null  int64  
 1   Date                                   14620 non-null  int64  
 2   number of bedrooms                     14620 non-null  int64  
 3   number of bathrooms                    14620 non-null  float64
 4   living area                            14620 non-null  int64  
 5   lot area                               14620 non-null  int64  
 6   number of floors                       14620 non-null  float64
 7   waterfront present                     14620 non-null  int64  
 8   number of views                        14620 non-null  int64  
 9   condition of the house                 14620 non-null  int64  
 10  grade of the house                     14620 non-null  int64  
 11  Ar

In [4]:
df.describe()

,id,Date,number of bedrooms,number of bathrooms,living area,lot area,number of floors,waterfront present,number of views,condition of the house,...,Built Year,Renovation Year,Postal Code,Lattitude,Longitude,living_area_renov,lot_area_renov,Number of schools nearby,Distance from the airport,Price
count,1.462000e+04,14620.000000,14620.000000,14620.000000,14620.000000,1.462000e+04,14620.000000,14620.000000,14620.000000,14620.000000,...,14620.000000,14620.000000,14620.000000,14620.000000,14620.000000,14620.000000,14620.000000,14620.000000,14620.000000,1.462000e+04
mean,6.762821e+09,42604.538646,3.379343,2.129583,2098.262996,1.509328e+04,1.502360,0.007661,0.233105,3.430506,...,1970.926402,90.924008,122033.062244,52.792848,-114.404007,1996.702257,12753.500068,2.012244,64.950958,5.389322e+05
std,6.237575e+03,67.347991,0.938719,0.769934,928.275721,3.791962e+04,0.540239,0.087193,0.766259,0.664151,...,29.493625,416.216661,19.082418,0.137522,0.141326,691.093366,26058.414467,0.817284,8.936008,3.675324e+05
min,6.762810e+09,42491.000000,1.000000,0.500000,370.000000,5.200000e+02,1.000000,0.000000,0.000000,1.000000,...,1900.000000,0.000000,122003.000000,52.385900,-114.709000,460.000000,651.000000,1.000000,50.000000,7.800000e+04
25%,6.762815e+09,42546.000000,3.000000,1.750000,1440.000000,5.010750e+03,1.000000,0.000000,0.000000,3.000000,...,1951.000000,0.000000,122017.000000,52.707600,-114.519000,1490.000000,5097.750000,1.000000,57.000000,3.200000e+05
50%,6.762821e+09,42600.000000,3.000000,2.250000,1930.000000,7.620000e+03,1.500000,0.000000,0.000000,3.000000,...,1975.000000,0.000000,122032.000000,52.806400,-114.421000,1850.000000,7620.000000,2.000000,65.000000,4.500000e+05
75%,6.762826e+09,42662.000000,4.000000,2.500000,2570.000000,1.080000e+04,2.000000,0.000000,0.000000,4.000000,...,1997.000000,0.000000,122048.000000,52.908900,-114.315000,2380.000000,10125.000000,3.000000,73.000000,6.450000e+05
max,6.762832e+09,42734.000000,33.000000,8.000000,13540.000000,1.074218e+06,3.500000,1.000000,4.000000,5.000000,...,2015.000000,2015.000000,122072.000000,53.007600,-113.505000,6110.000000,560617.000000,3.000000,80.000000,7.700000e+06


In [5]:
df.isnull().sum()

,0
id,0
Date,0
number of bedrooms,0
number of bathrooms,0
living area,0
lot area,0
number of floors,0
waterfront present,0
number of views,0
condition of the house,0


## **Missing Values Handle**



In [6]:
df.fillna(df.mean(numeric_only=True), inplace=True)

In [7]:
df = df.dropna()

## ***Q1. Average, Median and Standard Deviation of House Prices***

---



In [8]:
print("Average Price =", df['Price'].mean())
print("Median Price =", df['Price'].median())
print("Standard Deviation =", df['Price'].std())

Average Price = 538932.2183310534
Median Price = 450000.0
Standard Deviation = 367532.3808039669


## ***Interpretation***

### *The average house price is higher than the median price, which indicates that a few very expensive houses are increasing the overall average. This suggests the presence of high-value properties in the dataset.*



## ***Q2. Which number of bedrooms is most common?***

In [9]:
df['number of bedrooms'].mode()

,number of bedrooms
0,3


## ***Interpretation***

### *The most common house type has 3 bedrooms. This shows that 3-bedroom houses are the most preferred or most frequently available properties in the dataset.*

## ***Q3. Check if the price data is skewed***

In [10]:
df['Price'].skew()

np.float64(4.269297720707116)

## ***Interpretation***

### *The price distribution is positively skewed (right-skewed). Most houses have relatively lower prices, while a small number of houses have exceptionally high prices.*

## ***Q4. Average price for each number of bedrooms***

In [11]:
df.groupby('number of bedrooms')['Price'].mean()

,Price
number of bedrooms,
1,3.089638e+05
2,3.985476e+05
3,4.632776e+05
4,6.361988e+05
5,7.752550e+05
6,8.375815e+05
7,1.016544e+06
8,1.208455e+06
9,7.766663e+05


## ***Interpretation***

### *The average house price increases as the number of bedrooms increases. Larger houses generally have higher market values.*

## ***Q5. Relationship between Living Area and Price***

### ***(Use: Area of the house excluding basement)***

In [12]:
df['Area of the house(excluding basement)'].corr(df['Price'])

np.float64(0.6152204198678108)

## ***Interpretation***

### *There is a moderate positive relationship between living area and house price. Houses with larger living areas tend to have higher prices.*

## ***Q6. Identify anomalies where houses have high prices but low areas***

In [14]:
anomalies = df[
    (df['Price'] > df['Price'].quantile(0.90)) &
    (df['Area of the house(excluding basement)'] <
     df['Area of the house(excluding basement)'].quantile(0.25))
]
print(len(anomalies))

11


## ***Interpretation***

### *A few houses have unusually high prices despite having smaller living areas. These properties may be located in premium locations or may include special features that increase their value.*

## ***Q7. Compare average price based on number of floors and waterfront***

In [15]:
df.groupby(['number of floors','waterfront present'])['Price'].mean()

number of floors  waterfront present
1.0               0                     4.323724e+05
                  1                     1.211081e+06
1.5               0                     5.476162e+05
                  1                     1.232507e+06
2.0               0                     6.357742e+05
                  1                     2.075931e+06
2.5               0                     1.087654e+06
                  1                     3.280000e+06
3.0               0                     5.558095e+05
                  1                     1.655000e+06
3.5               0                     1.205875e+06
Name: Price, dtype: float64

## ***Interpretation***

### *Houses with waterfront views have significantly higher average prices than houses without waterfront views. Waterfront location has a strong impact on property value.*

## ***Q8. Minimum and Maximum House Price***

In [16]:
print(df['Price'].min())
print(df['Price'].max())

78000
7700000


## ***Interpretation***

### *The large difference between the minimum and maximum house prices indicates a wide variation in property values within the dataset.*

## ***Q9. Which location (zipcode/postal code) has the highest average price?***

In [17]:
df.groupby('Postal Code')['Price'].mean().sort_values(ascending=False).head(1)

,Price
Postal Code,
122071,2.348311e+06


## ***Interpretation***

### *The postal code with the highest average house price represents the most expensive area in the dataset. Properties in this location are generally valued much higher than in other areas.*

## ***Q10. Final Insights***

---
## ***Interpretation***

### *The analysis shows that factors such as number of bedrooms, living area, waterfront presence, and location strongly influence house prices. These variables can be useful for predicting and understanding property values.*


##**Insight 1**

####*3-bedroom houses are the most common type of house.*

##**Insight 2**

####*House prices are highly right-skewed because a few luxury properties have very high prices.*

##**Insight 3**

####*The average house price increases as the number of bedrooms increases.*

##**Insight 4**

####*Living area has a positive relationship with house price (correlation ≈ 0.615).*

##**Insight   5**

####*Waterfront properties are significantly more expensive than non-waterfront properties.*

##**Insight 6**

####*Some houses have unusually high prices despite having smaller areas, indicating the importance of location and premium features.*



